core/src/validator.rs

```diff
@@ -1661,12 +1662,26 @@ fn cleanup_accounts_path(account_path: &std::path::Path) {
 }
 
 pub fn is_snapshot_config_valid(
```

Comment:
This makes me wish that:
* `accounts_hash_interval_slots` remains unchanged
* `incremental_snapshot_interval` becomes in interval of `accounts_hash_interval_slots`
* `full_snapshot_interval` becomes in interval of `incremental_snapshot_interval`
then `is_snapshot_config_valid()` could be deleted as the config is always valid.

Reply:
That'd be a larger change though, if you want to punt on that I'd be fine with that

Reply:
Yeah, I like that. Is that something to change for here/now though?

If yes, there's somethings I think would need to be figured out first:

1. How would the cli arg names look? Would there be new args? What would happen to the old ones?
2. How would someone indicate "no snapshots"? This one I think is pretty easy with a new `--no-snapshots` flag.

Also, I think the one difference is that right now the full snapshot interval doesn't have to be a multiple of the incremental snapshot interval. It _could_ be, but the logic later doesn't require that.

Reply:
> Is that something to change for here/now though?

Na doesn't need to be here. However if we do want to go with that scheme then we'd want to replace the new args that are being introduced in this PR, fortunately there's no backwards compatibility requirement until master forks off to a release branch.

> I think the one difference is that right now the full snapshot interval doesn't have to be a multiple of the incremental snapshot interval. It could be, but the logic later doesn't require that.

Ah, right.  `full_snapshot_interval only` needs to be an interval of `accounts_hash_interval_slots`

Reply:
Up to you if you want to pick that nit though, the current approach works well enough IMO

Reply:
I wonder what's more intuitive for the user/validator. Maybe keeping the interval in slots is the way to go...

I was thinking of how the help text would display this info, and how default values would show up. Also, because hopefully everyone uses the same full snapshot interval. So then if the user changes the accounts hash interval, that would change the default value for a full snapshot interval multiplier to hit the same interval in slots...

Bleh, these args are too intertwined, I think, to have an obvious "best" answer. With that, my plan is to leave the args as-is (assuming that's still OK).